File: test5empty.py
import requests
import json

# Your Västtrafik credentials
client_id = ''
client_secret = ''
auth_url = 'https://ext-api.vasttrafik.se/token'
api_url = 'https://ext-api.vasttrafik.se/pr/v4/journeys'

# Step 1: Get an access token
def get_access_token(client_id, client_secret):
    auth = (client_id, client_secret)
    data = {'grant_type': 'client_credentials'}
    response = requests.post(auth_url, auth=auth, data=data)

    if response.status_code == 200:
        return response.json().get('access_token')
    else:
        print(f"Error fetching access token: {response.status_code} {response.text}")
        return None

# Step 2: Make a request to the /journeys endpoint
def get_journey(access_token, origin_gid, destination_gid, date, time):
    headers = {'Authorization': f'Bearer {access_token}'}
    params = {
        'originGid': origin_gid,
        'destinationGid': destination_gid,
        'date': date,      # Format: YYYY-MM-DD
        'time': time,      # Format: HH:MM
        'searchForArrival': False  # True for arrival, False for departure
    }

    response = requests.get(api_url, headers=headers, params=params)

    if response.status_code == 200:
        return response.json()
    else:
        print(f"Error making /journeys request: {response.status_code} {response.text}")
        return None

# Save the result to a JSON file
def save_to_json(data, filename):
    with open(filename, 'w', encoding='utf-8') as json_file:
        json.dump(data, json_file, ensure_ascii=False, indent=4)
    print(f"Result saved to {filename}")

# Example usage
if __name__ == '__main__':
    token = get_access_token(client_id, client_secret)
    if token:
        # Replace these with actual GIDs from Västtrafik's system
        origin_gid = '9021014001760000'  # Example: Gothenburg Central Station
        destination_gid = '9021014003980000'  # Example: Another valid GID
        date = '2024-12-10'  # Specify your desired date
        time = '14:00'       # Specify your desired time

        journey_data = get_journey(token, origin_gid, destination_gid, date, time)
        if journey_data:
            save_to_json(journey_data, 'journey_result.json')

File: test5empty.js
const fs = require('fs');

// Your Västtrafik credentials
const clientId = '';
const clientSecret = '';
const authUrl = 'https://ext-api.vasttrafik.se/token';
const apiUrl = 'https://ext-api.vasttrafik.se/pr/v4/journeys';

// Function to get an access token
async function getAccessToken(clientId, clientSecret) {
    const credentials = Buffer.from(`${clientId}:${clientSecret}`).toString('base64');
    const body = new URLSearchParams({ grant_type: 'client_credentials' });

    try {
        const response = await fetch(authUrl, {
            method: 'POST',
            headers: {
                'Authorization': `Basic ${credentials}`,
                'Content-Type': 'application/x-www-form-urlencoded',
            },
            body: body.toString(),
        });

        if (response.ok) {
            const data = await response.json();
            return data.access_token;
        } else {
            console.error(`Error fetching access token: ${response.status} ${await response.text()}`);
            return null;
        }
    } catch (error) {
        console.error('Error fetching access token:', error);
        return null;
    }
}

// Function to make a request to the /journeys endpoint
async function getJourney(accessToken, originGid, destinationGid, date, time, mode) {
    const params = new URLSearchParams({
        originGid,
        destinationGid,
        date,  // Format: YYYY-MM-DD
        time,  // Format: HH:MM
        searchForArrival: 'false',  // true for arrival, false for departure
        transportModes: mode,
    });

    try {
        const response = await fetch(`${apiUrl}?${params.toString()}`, {
            method: 'GET',
            headers: {
                'Authorization': `Bearer ${accessToken}`,
            },
        });

        if (response.ok) {
            return await response.json();
        } else {
            console.error(`Error making /journeys request: ${response.status} ${await response.text()}`);
            return null;
        }
    } catch (error) {
        console.error('Error making /journeys request:', error);
        return null;
    }
}

// Function to save the result to a JSON file
function saveToJson(data, filename) {
    try {
        fs.writeFileSync(filename, JSON.stringify(data, null, 4), 'utf-8');
        console.log(`Result saved to ${filename}`);
    } catch (error) {
        console.error(`Error saving file: ${error.message}`);
    }
}

// Example usage
(async () => {
    const token = await getAccessToken(clientId, clientSecret);
    if (token) {
        // Replace these with actual GIDs from Västtrafik's system
        const originGid = '9021014003980000';  // Example: Gothenburg Central Station
        const destinationGid = '9021014001050000';  // Example: Another valid GID
        const date = '2024-12-10';  // Specify your desired date
        const time = '14:00';       // Specify your desired time
        const mode = 'walk';

        const journeyData = await getJourney(token, originGid, destinationGid, date, time, mode);
        if (journeyData) {
            saveToJson(journeyData, 'journey_resultJS.json');
        }
    }
})();
